libs/vm-modules/src/math/tensor/tensor.cpp

```diff
@@ -195,12 +195,31 @@ Ptr<VMTensor> VMTensor::Unsqueeze()
 
 bool VMTensor::Reshape(Ptr<Array<SizeType>> const &new_shape)
 {
+  if (new_shape->elements.empty())
+  {
+    RuntimeError("Can not reshape a Tensor : new shape is empty!");
+    return false;
+  }
+  for (SizeType axis_size : new_shape->elements)
+  {
+    if (axis_size == 0)
+    {
+      RuntimeError("Can not reshape a Tensor : axis of size 0 found in new shape!");
+      return false;
+    }
+  }
```

Comment:
you don't check that number of elements in new_shape are the same as in old_shape. As Ira pointed out, this would be necessary to match numpy/tf behaviour - and we should probably match their behaviour if it's easy to do

Reply:
well, here I just checked an obviously bad cases, but did not change the existing "good" behaviour. I agree with Ira that silent dropping of a tensor content while reshaping is not the most clear and user-friendly approach; however, I don't consider myself as enough skilled ML engineer to decide what the correct behavior is :) I would suggest discussing that with ML team and/or requirement management.

Reply:
UPD: VMTensor behavior could safely be changed without affecting underlying math::Tensor implementation, so I added element count equality check here.